scene/animation/tween.cpp

```diff
@@ -1203,10 +1216,40 @@ bool Tween::_calc_delta_val(const Variant &p_initial_val, const Variant &p_final
 			delta_val = Color(f.r - i.r, f.g - i.g, f.b - i.b, f.a - i.a);
 		} break;
 
-		default:
-			// TODO: Should move away from a 'magic string'?
-			ERR_PRINT("Invalid param type, except(int/real/vector2/vector/matrix/matrix32/quat/aabb/transform/color)");
+		case Variant::RECT2: {
+			// Build a new Rect2 and use the new position and sizes to make a delta
+			Rect2 i = initial_val;
+			Rect2 f = final_val;
+			delta_val = Rect2(f.position - i.position, f.size - i.size);
+		} break;
+
+		default: {
+			static Variant::Type supported_types[] = {
+				Variant::BOOL,
+				Variant::INT,
+				Variant::REAL,
+				Variant::VECTOR2,
+				Variant::VECTOR3,
+				Variant::BASIS,
+				Variant::TRANSFORM,
+				Variant::TRANSFORM2D,
+				Variant::AABB,
+				Variant::COLOR,
+				Variant::RECT2,
+			};
+
+			int length = *(&supported_types + 1) - supported_types;
+			String error_msg = "Invalid param type, excepting ";
```

Comment:
That's still wrong English, I'd use instead:
```
Invalid parameter type. Supported types are: bool, int, float, ..., Color, Rect2.
```
Also please make sure to use the same ordering as in `variant.h` (including in the `case`s).

Otherwise it looks great :)